provider/plugin/plugin.go

```diff
@@ -0,0 +1,350 @@
+	if err != nil {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("Failed to perform request: %s", err.Error())
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("Failed to apply changes with code %d", resp.StatusCode)
+		return fmt.Errorf("failed to apply changes with code %d", resp.StatusCode)
+	}
+	return nil
+}
+
+// PropertyValuesEqual will call the provider doing a POST on `/propertyvaluesequal` which will return a boolean in the format
+// `{propertyvaluesequal: true}`
+// Errors in anything technically happening from the provider will return true so that no update is performed.
+// Errors will also be logged and exposed as metrics so that it is possible to alert on them if needed.
+func (p PluginProvider) PropertyValuesEqual(name string, previous string, current string) bool {
+	u, err := url.JoinPath(p.remoteServerURL.String(), "propertyvaluesequal")
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("Failed to join path: %s", err)
+		return true
+	}
+	b, err := json.Marshal(&PropertyValuesEqualRequest{
+		Name:     name,
+		Previous: previous,
+		Current:  current,
+	})
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("Failed to marshal request: %s", err)
+		return true
+	}
+
+	req, err := http.NewRequest("POST", u, bytes.NewBuffer(b))
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("Failed to create request: %s", err)
+		return true
+	}
+	req.Header.Set(acceptHeader, mediaTypeFormatAndVersion)
+	resp, err := p.client.Do(req)
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("Failed to perform request: %s", err)
+		return true
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("Failed to run PropertyValuesEqual with code %d", resp.StatusCode)
+		return true
+	}
+
+	respoBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Errorf("Failed to read body: %v", err)
+		return true
+	}
+	r := PropertyValuesEqualResponse{}
+	err = json.Unmarshal(respoBody, &r)
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Errorf("Failed to unmarshal body: %v", err)
+		return true
+	}
+	return r.Equals
+}
+
+// AdjustEndpoints will call the provider doing a POST on `/adjustendpoints` which will return a list of modified endpoints
+// based on a provider specific requirement.
+// This method returns an empty slice in case there is a technical error on the provider's side so that no endpoints will be considered.
+func (p PluginProvider) AdjustEndpoints(e []*endpoint.Endpoint) []*endpoint.Endpoint {
+	endpoints := []*endpoint.Endpoint{}
+	u, err := url.JoinPath(p.remoteServerURL.String(), "adjustendpoints")
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("Failed to join path, %s", err)
+		return endpoints
+	}
```

Comment:
```suggestion
	u := p.remoteServerURL.JoinPath("adjustendpoints").String()
```
see [this comment](https://github.com/kubernetes-sigs/external-dns/pull/3063/files/bdc19db88b1a9793cc2e2473e1c932673709112d#r1224361705)